### DISCUSSION TITLE
Includes to be checked in the library list

### DISCUSSION THREAD

**Discussion by rishabkshk** [codefori/vscode-ibmi#1188] [Q&A]
Hi,

I have my program member and copy source members in different libraries. I am able to access the includes(view implementation) if I specify the full include path e.g.` /copy srclib/srcfile, ctlopts ` and not with ` /copy srcfile, ctlopts `. I have tried to search the settings but did not find any option for RPGLE content assist to check in the library list I have setup for Code for IBM i.

Is there a way for the RPGLE content assist feature to pick the includes from a library other than actual program member's without hardcoding the include's library?

**Comment by SJLennon** [comment]
Perhaps put your copy member source library at the top of the library list? The compiler supports this.

IBM docs say this:

/COPY or /INCLUDE followed by exactly one space followed by the file name or path
when specifying a physical file, the library, file, and member name, can be in one of these formats:
libraryname/filename,membername
filename,membername
membername

A member name must be specified.
If a file name is not specified, QRPGLESRC is assumed.
If a library is not specified, the library list is searched for the file. All occurrences of the specified source file in the library list are searched for the member until it is located or the search is complete.
If a library is specified, a file name must also be specified.

[https://www.ibm.com/docs/en/i/7.4?topic=directives-copy-include](url)

**Reply by rishabkshk** [reply]
Yes the compilation works fine, it's just the rpgle content assist not able to use the library list to search for includes. It just checks the library in which the main program source is. I just want the procedure and ds declarations that are in my copybooks to be available(as a hint/autocomplete) in the main source. Also I should be able to use the `view implementation` functionality.

**Reply by Gregory1701** [reply]
@worksofliam @sebjulliand @chrjorgensen Any thoughts on this?  Can we get someone working on this please?

**Reply by worksofliam** [reply]
@Gregory1701 pull request put into vscode-rpgle today for source members resolving via the library list!

**Reply by Gregory1701** [reply]
@worksofliam Awesome, thanks!

**Reply by Gregory1701** [reply]
@worksofliam Is this change live?

Also wondering if this change will make referencing a copy book with *libl, ex. /copy *libl/qrpglesrc,mycopybook work.

**Reply by worksofliam** [reply]
Not's not live yet (still pending some changes after a review). I am hoping for next week.

Also, I don't think the `*LIBL` keyword is valid in a copy or include directive: https://www.ibm.com/docs/en/i/7.1?topic=directives-copy-include

If no library is specified, it will use the library list.

Can you share some docs where that's referenced?

**Reply by Gregory1701** [reply]
Okay, thanks.

I don't find any documentation talking about using *libl, but pretty confident we specified that all the time in my previous RPG shop.  I did find that '*libl' is assumed if no library is specified.

**Reply by worksofliam** [reply]
@Gregory1701 I will make sure you know when it's been released :)

**Reply by Gregory1701** [reply]
Cool, thanks.

**Reply by worksofliam** [reply]
If you want, you're able to build and test the PR here: https://github.com/halcyon-tech/vscode-rpgle/pull/225
